.github/workflows/shader.yml

```diff
@@ -0,0 +1,13 @@
+on:
+  pull_request:
+    branches-ignore:
+     - main
+
+jobs:
+  push-shaders:
+    runs-on: windows-latest
+    name: compile shaders
+    steps:
+      - uses: actions/checkout@v3
+      - run: mkdir piet-gpu/shader/gen
```

Comment:
Adding `-force` here should suppress the error if the directory already exists. It might be worth not doing this though and just rejecting any PRs that contain the gen directory?